UM/Settings/AdditionalSettingDefinitionAppender.py

```diff
@@ -0,0 +1,108 @@
+
+    Appended settings may not be the same as 'baseline' (so any 'non-appended' settings) settings.
+    (But may of course clash between different providers and versions, that's the whole point of this function...)
+    Furthermore, it's assumed that formulas within the appended settings will only use settings either;
+     - as defined within the baseline, or;
+     - any other settings defined _by the provider itself_.
+
+    For each key that is renamed, this results in a mapping <key> -> _<provider_type>__<id*>__<version>__<key>
+     where '<id*>' is the version of the provider, but converted from using points to using underscores.
+    Example: 'tapdance_factor' might become '_plugin__dancingprinter__1_2_99__tapdance_factor'
+    Also note that all the tag_... parameters will be forced to lower-case.
+
+    :param tag_type: Type of the additional settings appender, for example; "PLUGIN".
+    :param tag_id: ID of the provider. Should be unique.
+    :param tag_version: Version of the provider. Points will be replaced by underscores.
+    :param settings: The settings as originally provided.
+
+    :returns: Remapped settings, where each settings-name is properly tagged/'namespaced'.
+    """
+    tag_type = tag_type.lower()
+    tag_id = tag_id.lower()
+    tag_version = tag_version.lower().replace(".", "_")
+
+    # First get the mapping, so that both the 'headings' and formula's can be renamed at the same time later.
+    def _getMapping(values: Dict[str, Any]) -> Dict[str, str]:
+        result = {}
+        for key, value in values.items():
+            mapped_key = key
+            if isinstance(value, dict):
+                if "type" in value and value["type"] != "category":
+                    mapped_key = f"_{tag_type}__{tag_id}__{tag_version}__{key}"
+                result.update(_getMapping(value))
+            result[key] = mapped_key
+        return result
+    key_map = _getMapping(settings)
+
+    # Get all values that can be functions, so it's known where to replace.
+    function_type_names = set(SettingDefinition.getPropertyNames(DefinitionPropertyType.Function))
+
+    # Replace all, both as key-names and their use in formulas.
+    def _doReplace(values: Dict[str, Any]) -> Dict[str, str]:
+        result = {}
+        for key, value in values.items():
+            if key in function_type_names and isinstance(value, str):
+                # Replace key-names in the specified settings-function.
+                for original, mapped in key_map.items():
+                    value = value.replace(original, mapped)
+            elif isinstance(value, dict):
+                # Replace key-name 'heading'.
+                key = key_map.get(key, key)
+                value = _doReplace(value)
+            result[key] = value
+        return result
+    return _doReplace(settings)
+
+
+class AdditionalSettingDefinitionsAppender(PluginObject):
+
```

Comment:
What is an additional settings defintions appender? When should one use it? (aka: I think that this needs documentation)